src/System.Linq.Expressions/src/System/Runtime/CompilerServices/ReadOnlyCollectionBuilder.cs

```diff
@@ -23,7 +25,7 @@ public sealed class ReadOnlyCollectionBuilder<T> : IList<T>, System.Collections.
         private int _version;
 
         [NonSerialized]
-        private Object _syncRoot;
+        private object _syncRoot;
```

Comment:
Perhaps we are concerned that somebody is using both List instance and its SyncObject for locking at the same time and expect them to be different...
I'd find that extremely unlikely.